exercises/all-your-base/example.rs

```diff
@@ -31,15 +38,18 @@ pub type Digit = u32;
 ///  * Never output leading 0 digits. However, your function must be able to
 ///     process input with leading 0 digits.
 ///
-pub fn convert<P: AsRef<[Digit]>>(digits: P, from_base: Digit, to_base: Digit) -> Result<Vec<Digit>, &'static str> {
+pub fn convert<P: AsRef<[Digit]>>(digits: P, from_base: Digit, to_base: Digit) -> Result<Vec<Digit>, Error> {
     // check that both bases are in the correct range
-    if from_base < 2 || to_base < 2 {
-        return Err("Invalid base");
+    if from_base < 2 {
+        return Err(Error::InvalidInputBase);
+    }
+    if to_base < 2 {
+        return Err(Error::InvalidOutputBase);
     }
 
     // check that all digits are in the correct range specified by the base
-    if digits.as_ref().iter().any(|&num| num >= from_base) {
-        return Err("Digits invalid for input base");
+    if let Some(&invalid) = digits.as_ref().iter().find(|&num| *num >= from_base) {
```

Comment:
There is in fact not a specific reason I wrote `find(|&num| *num >= from_base)` as opposed to one of these two alternatives:

```
find(|&&num| num >= from_base)
find(|num| **num >= from_base)
```

I really don't know what I'm doing and which is better, so I need help on that.

Reply:
I am also unaware of any particular style guide suggestion as to which of those alternatives is ideal, but I like the symmetry of having equal numbers of `&` and `*`.